climpred/prediction.py

```diff
@@ -28,6 +35,81 @@
 )
 
 
+def get_metric_comparison_dim(metric, comparison, dim, kind):
+    """Check whether `metric`, `comparison` and `dim` can work together in experiment
+    type `kind` and return corrected.
+
+    Args:
+        metric (str): metric or alias string
+        comparison (str): Description of parameter `comparison`.
+        dim (list of st or str): dimension to apply metric to.
+        kind (str): experiment type from ['hindcast', 'PM'].
+
+    Returns:
+        metric (Metric): metric class
+        comparison (Comparison): comparison class.
+        dim (list of st or str): corrected dimension to apply metric to.
+    """
+    # check kind allowed
+    is_in_list(kind, ['hindcast', 'PM'], str)
+    # set default dim
+    if dim is None:
+        dim = 'init' if kind == 'hindcast' else ['init', 'member']
+    # check allowed dims
+    if kind == 'hindcast':
+        is_in_list(dim, ['member', 'init'], str)
+    elif kind == 'PM':
+        is_in_list(dim, ['member', 'init', ['init', 'member']], '')
+
+    # get metric and comparison strings incorporating alias
+    metric = METRIC_ALIASES.get(metric, metric)
+    comparison = COMPARISON_ALIASES.get(comparison, comparison)
+
+    METRICS = HINDCAST_METRICS if kind == 'hindcast' else PM_METRICS
+    COMPARISONS = HINDCAST_COMPARISONS if kind == 'hindcast' else PM_COMPARISONS
+    # get class from string metric(Metric)
```

Comment:
I think this and the following comment can go since the function name is self-explanatory